nssp_demo/collate_score_tables.R

```diff
@@ -0,0 +1,212 @@
+library(data.table)
+library(argparser)
+
+#' Get all the subdirectories within a parent directory
+#' that match the pattern for a forecast run for a
+#' given disease and optionally a given report date.
+#'
+#' @param parent_dir Directory in which to look for forecast subdirectories.
+#' @param diseases Names of the diseases to match, as a vector of strings,
+#' or a single disease as a string.
+#' @return A vector of paths to the forecast subdirectories.
+get_all_forecast_dirs <- function(dir_of_forecast_date_dirs,
+                                  diseases) {
+  # disease names are lowercase by convention
+  match_patterns <- stringr::str_c(tolower(diseases), "_r")
+  matches_diseases <- \(x) {
+    any(stringr::str_starts(x, pattern = match_patterns))
+  }
+
+  dirs <- tibble::tibble(
+    dir_path =
+      fs::dir_ls(dir_of_forecast_date_dirs,
+        type = "directory"
+      )
+  ) |>
+    dplyr::mutate(dir_name = fs::path_file(dir_path)) |>
+    dplyr::filter(purrr::map_lgl(dir_name, matches_diseases)) |>
```

Comment:
This could be done with a `filter(str_starts(path_file(dir_name), str_c(str_to_lower(diseases), collapse = "|")))`

Reply:
Edited in ca93bf2